code-studio/src/js/components/GridEditor.jsx

```diff
@@ -101,39 +111,86 @@ var GridEditor = React.createClass({
     // Both of those seem a bit unnecessary, so for now this hack will
     // remain.
     var cells = this.state.cells;
-    var newCell = this.getCellClass().deserialize(newSerializedCell);
-    if (row !== undefined && col !== undefined) {
-      cells[row][col] = newCell;
-    }
+    newCells.forEach(function (_, i) {
+      _.forEach(function (cell, j) {
+        if (cells[row + i] && cells[row + i][col + j]) {
+          cells[row + i][col + j] = this.getCellClass().deserialize(cell);
+        }
+      }, this);
+    }, this);
+
     var serializedData = cells.map(function (row) {
       return row.map(function (cell) {
         return cell.serialize();
       });
     });
+
     this.props.onUpdate(serializedData);
     this.setState({
       cells: cells
     });
   },
 
+  /**
+   * When a given cell is modified, update the grid
+   */
+  handleCellChange: function (newSerializedCell) {
+    var row = this.state.selectedRow;
+    var col = this.state.selectedCol;
+    // updateCells expects a two-dimentional array
+    this.updateCells(row, col, [[newSerializedCell]]);
+  },
+
+  /**
+   * "Paste" the cells in our "clipboard" into the grid
+   */
+  pasteCopiedCells: function () {
+    var copiedCells = this.state.copiedCells;
+    var row = this.state.selectedRow;
+    var col = this.state.selectedCol;
+    this.updateCells(row, col, copiedCells);
+  },
+
+  /**
+   * Store the given cells on our "clipboard"
+   */
+  setCopiedCells: function (cells) {
+    this.setState({
+      copiedCells: cells
+    });
+  },
+
   render: function () {
     var cells = this.state.cells;
 
     var cellEditor;
     var selectedCellJson;
+    var pasteButton;
     var row = this.state.selectedRow;
     var col = this.state.selectedCol;
     if (cells[row] && cells[row][col]) {
       var cell = cells[row][col];
       var EditorClass = this.getEditorClass();
       cellEditor = <EditorClass cell={cell} row={row} col={col} onUpdate={this.handleCellChange} />;
       selectedCellJson = <CellJSON serialization={cell.serialize()} onChange={this.handleCellChange} />;
+      if (this.state.copiedCells) {
+        pasteButton = (<button type="button" onClick={this.pasteCopiedCells}>
+            {"Paste Selected " + this.state.copiedCells.length + "x" + this.state.copiedCells[0].length + " Cells Excel-Style"}
```

Comment:
I'd omit "Excel-Style" here, if only to minimize expectations of future feature-parity :stuck_out_tongue_winking_eye: